src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -560,50 +577,37 @@ private void processLogsAndUpload() throws InterruptedException {
             }
             AtomicReference<Optional<ComponentLogFileInformation>> componentLogFileInformation =
                     new AtomicReference<>(Optional.empty());
+            AtomicBoolean isActiveFileCompleted = new AtomicBoolean(false);
```

Comment:
If this is all running on the same thread there is no need to use an `AtomicBoolean`